### PR TITLE
Update branch protection

### PR DESCRIPTION
**Pull request by fejta**
* Mergify repos should have at least one approval from code owners
* Tide repos should not use code owners (yet)
* Require two approvals for api
* Outline how to restrict merges to release teams (do these teams exist yet?)
* Unit tests to cover each scenario.

### CODE REVIEW

File: prow/config.yaml
Comment (geeknoid) [inline prow/config.yaml]: This will break how our community uses branches. There are non-standard branches in  our repos which should not have branch protection enabled. We only want master and release-* to have branch protection.

Comment (fejta) [reply prow/config.yaml]: 1) What does this help? What does it accomplish that they cannot do with a fork?
2) If these are useful why shouldn't they have branch protection?
3) If they are small in number, we can specifically exclude them (like we do in kubernetes)?

**Comment by geeknoid** [comment]
The branches are created in an ad hoc fashion. For example, in istio.io you
get branches created automatically when you select "Edit this Page" on the
main Istio web site.

Folks create branches against a variety of repos. They could indeed achieve
the same effect with a fork, and we encourage them to do so.  But the fact
is that manh of our developers create branches on our main repos. I
regularly go through the repos and delete old merged branches to keep
things clean.

I think  turning on branch protection for custom branches will break our
existing workflows.

**Comment by fejta** [comment]
1) We have the ability to shape and influence social norms in addition to automated behavior.
2) We can automate the behavior we want to encode
3) This sounds like that might be a good reason to special case the istio.io repo to allow unprotected repos by default

**Comment by howardjohn** [comment]
Regarding 3, I just did the "edit through the ui" option and it automagically created a fork for me (see https://github.com/howardjohn/mergify-test/pull/13)

**Comment by fejta** [comment]
The revert button does the same thing unfortunately

**Comment by fejta** [comment]
A branch-cleanup script would be interesting/useful -- aka delete any branch that does/doesn't match a list and hasn't been updated in some time-period.

**Comment by geeknoid** [comment]
So given that we want consistent branch protection for all our "master" and "release-X.Y" branches across repos, but specifically want no protection for other branches in those same repos, how should we proceed?

**Comment by geeknoid** [comment]
Do we still need this PR?

**Comment by fejta** [comment]
/hold cancel
@geeknoid PTAL this configures branch protection to most of what we talked about (require code owners except where tide is used), 2 approvals for api

**Comment by geeknoid** [comment]
This looks good.

Thanks.

**Review by geeknoid — APPROVED** [review]

**Pull request merged**
